test/parallel_api/iterator/iterators.pass.cpp

```diff
@@ -308,6 +308,10 @@ struct test_permutation_iterator
         EXPECT_TRUE(perm_it1.map() == in2.begin(), "wrong result from permutation_iterator::map");
 
         test_random_iterator(perm_begin);
+
+        auto n = in1.size();
+        auto perm_it_fun_rev = oneapi::dpl::make_permutation_iterator(in1.begin(), [n] (auto i) { return n - i - 1;});
+        EXPECT_TRUE(*++perm_it_fun_rev == *(in1.end()-2), "wrong result from permutation_iterator(base, functor)");
     }
 };
 
```

Comment:
May be would be good also to add example from https://www.boost.org/doc/libs/1_38_0/libs/iterator/doc/permutation_iterator.html
and use it like our test?

Reply:
Making the test coverage wider makes sense. But, I'm afraid we cannot add the example as is by legal reasons.

Reply:
BTW, i'm going to arrange API of permutation iterator like variant from boost library.